python/foglamp/plugins/north/pi_server/pi_server.py

```diff
@@ -799,28 +795,37 @@ def _generate_omf_typename_automatic(self, asset_code):
                         status_code = resp.status
                         text = await resp.text()
 
-            except Exception as e:
-                _error = Exception(plugin_common.MESSAGES_LIST["e000024"].format(e))
-                _message = plugin_common.MESSAGES_LIST["e000024"].format(e)
+            except Exception as ex:
+
+                # The conversion to a string of the exception TimeoutError produces an empty string,
+                # so it forces a proper message
+                if str(ex) == "":
+                    details = type(ex).__name__
+                else:
+                    details = str(ex)
+
+                _message = plugin_common.MESSAGES_LIST["e000024"].format(self._config['URL'], details)
+                _error = plugin_exceptions.URLConnectionError(_message)
             else:
                 # Evaluate the HTTP status codes
                 if not str(status_code).startswith('2'):
-                    tmp_text = str(status_code) + " " + text
-                    _message = plugin_common.MESSAGES_LIST["e000024"].format(tmp_text)
-                    _error = plugin_exceptions.URLFetchError(_message)
+                    _tmp_text = "status code " + str(status_code) + " - " + text
+                    _message = plugin_common.MESSAGES_LIST["e000024"].format(self._config['URL'], _tmp_text)
+                    _error = plugin_exceptions.URLConnectionError(_message)
 
                 self._logger.debug("message type |{0}| response: |{1}| |{2}| ".format(
-                                                                                message_type,
-                                                                                status_code,
-                                                                                text))
+                    message_type,
+                    status_code,
+                    text))
+
             if _error:
                 time.sleep(sleep_time)
```

Comment:
not realted to this  PR. but should this not be `await asyncio.sleep(sleep_time)`? This is sync code in async def and might be freezing event_loop

Reply:
you are right, fixed.